In [1]:
import polars as pl
import sys
from pathlib import Path
sys.path.append(str(Path("../scripts")))
from RQ4_plot_lib import plot_brand_vs_free_prices, plot_price_difference, plot_brand_comparison, plot_avg_premium_per_brand, save_png

csv_path_brandvsfree = Path("D:/Tankdaten/rq_brand_vs_free/brand_vs_free_yearly_stats.csv")
csv_per_brand = Path("D:/Tankdaten/rq_brand_vs_free/per_brand_yearly_stats.csv")

In [2]:
df_brandvsfree = pl.read_csv(csv_path_brandvsfree).filter(pl.col("year") < 2026) # Filter out 2026 data because it's incomplete
df_per_brand = pl.read_csv(csv_per_brand).filter(pl.col("year") < 2026) # Filter out 2026 data because it's incomplete

brand = df_brandvsfree.filter(pl.col("station_type") == "brand_station").sort("year")
free  = df_brandvsfree.filter(pl.col("station_type") == "free_station").sort("year")
# print(df_brandvsfree)
print(df_per_brand.tail(20))

shape: (20, 6)
┌──────────────────┬──────┬────────────────────┬─────────────┬──────────┬──────────┐
│ brand_normalized ┆ year ┆ price_update_count ┆ diesel_mean ┆ e5_mean  ┆ e10_mean │
│ ---              ┆ ---  ┆ ---                ┆ ---         ┆ ---      ┆ ---      │
│ str              ┆ i64  ┆ i64                ┆ f64         ┆ f64      ┆ f64      │
╞══════════════════╪══════╪════════════════════╪═════════════╪══════════╪══════════╡
│ TOTAL            ┆ 2018 ┆ 3758686            ┆ 1.286224    ┆ 1.457374 ┆ 1.437282 │
│ TOTAL            ┆ 2019 ┆ 4728621            ┆ 1.261102    ┆ 1.432604 ┆ 1.411881 │
│ TOTAL            ┆ 2020 ┆ 5472914            ┆ 1.109786    ┆ 1.290801 ┆ 1.254711 │
│ TOTAL            ┆ 2021 ┆ 6365465            ┆ 1.387535    ┆ 1.582151 ┆ 1.523271 │
│ TOTAL            ┆ 2022 ┆ 6370939            ┆ 1.94754     ┆ 1.913527 ┆ 1.854266 │
│ …                ┆ …    ┆ …                  ┆ …           ┆ …        ┆ …        │
│ WESTFALEN        ┆ 2021 ┆ 1490969            ┆ 1

## Mean Fuel Prices: Brand vs. Free Stations (per year)

In [3]:
plot_brand_vs_free_prices(brand, free)

Brand stations charge a **consistent** premium of ~2 ct/L across all fuel types and all years. The gap is stable even during the 2022 energy price spike, suggesting it is structural rather than market-driven.

## Price Difference: Brand minus Free (per year)

In [4]:
plot_price_difference(brand, free)

Diesel: avg difference = +1.753 ct/L
E5: avg difference = +1.620 ct/L
E10: avg difference = +1.525 ct/L


## Price Difference: ARAL - other

In [5]:
plot_brand_comparison(df_per_brand, df_brandvsfree, fuel="e10_mean", brands=["ARAL", "OIL!", "ESSO"])

**ARAL** is the most expensive brand in nearly **every year**, typically a few cents above both the brand average and free stations. **ESSO** briefly exceeded ARAL's prices in 2022–2023 during the energy crisis. **OIL!** stands out as an outlier — it consistently prices below even the average free station.

## Average Price Premium per Brand vs. Free Stations

In [6]:
fig = plot_avg_premium_per_brand(df_per_brand, free, fuel="e10_mean")
fig.show()

**ARAL** charges the highest average premium at around +3 ct/L above free stations — the largest gap among all brands. Most other brands cluster between 0 and +2 ct/L. Notably, several brands including OIL! show a **negative** premium, meaning they are on average cheaper than independent free stations.

## **Results**

This analysis compared fuel prices between **brand gas stations** (e.g. ARAL, ESSO, WESTFALEN) and **free/independent gas stations** in Germany from 2014 to 2025.

**Brand stations are consistently more expensive** than free stations across all fuel types and all years, with a average difference of:

- **Diesel:** +1.75 ct/L
- **E5:** +1.62 ct/L
- **E10:** +1.53 ct/L

The price gap remained remarkably stable over time at roughly **~2 ct/L**, even during the energy price spike in 2022.

Looking at individual brands, **ARAL is the most expensive brand** with an average premium of around **+3 ct/L** compared to free stations. **ESSO** briefly surpassed ARAL as the most expensive brand in 2022 and 2023. On the other end of the spectrum, **OIL!** is consistently the cheapest brand — even undercutting the average free station price throughout the entire observation period.

Overall, the results suggest that brand affiliation has a **small but persistent effect** on fuel prices, and that the brand premium is largely independent of broader market fluctuations.